### DISCUSSION TITLE
How to set a job to run when pushing to a PR branch?

### DISCUSSION THREAD

**Discussion by l0b0** [community/community#27179] [Actions]
<p>As far as I can tell <code>on: [pull_request]</code> is <em>not</em> the right thing, because that <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc">triggers when doing various other actions on a pull request</a>, such as requesting a review or closing it. <code>on: [push]</code> combined with <code>if: github.event_name == 'pull_request'</code> doesn’t seem to work. If I understand correctly, this is because they can’t both be true at the same time.</p>
<p>I’m starting to think I might’ve missed something, because there are heaps of people which seem to say that <code>on: [pull_request]</code> means “when pushing to a pull request branch”, but I can’t reconcile that with the documentation. I thought maybe the <code>synchronize</code> activity type might be related to pushes, but the official documentation doesn’t seem to have any details. <a href="https://frontside.com/blog/2020-05-26-github-actions-pull_request/" rel="noopener nofollow ugc">This article</a> seems to corroborate the common usage, but that’s the only one I could find.</p>
<p>This is related to another question (as a new user I’m not allowed to add any more links to this post) – basically I’m trying to figure out how to run jobs only when pushing to</p>
<ul>
<li>a branch</li>
<li>for which there exists a pull request</li>
<li>which is open and</li>
<li>which is not a draft.</li>
</ul>

**Comment by logankilpatrick** [comment]
<p>What you are looking for is:</p>
<pre><code class="lang-auto">on:
  pull_request:
    types: [opened, synchronize, reopened]
</code></pre>
<p>Where <code>synchronize</code> means a change has been pushed to the PR branch.</p>
<p>Edit, per the GitHub docs: <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request</a> it looks like this is actually the default behavior:</p>
<blockquote>
<p><strong>Note:</strong> By default, a workflow only runs when a <code>pull_request</code> 's activity type is <code>opened</code> , <code>synchronize</code> , or <code>reopened</code> . To trigger workflows for more activity types, use the <code>types</code> keyword.</p>
</blockquote>

**Comment by logankilpatrick** [comment, marked as answer]
<p>I do agree that the docs on this are not explicitly clear about. I had to look up what “synchronize” meant in this context. I just opened an issue to hopefully fix this: <a href="https://github.com/github/docs/issues/2257" rel="noopener nofollow ugc">https://github.com/github/docs/issues/2257</a></p>
